### PR TITLE
Refactor test suite

### LINKED ISSUE #4635: Create a partial RTL-based testing suite to be run in conjunction with existing test suite

### PR DESCRIPTION
**Pull request by drewbo**
## Changes proposed in this pull request:
- Close #4635 
- Adds RTL and jest structure
- Adds ~6 component tests

## security considerations
None

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,11 +32,13 @@
     "express-session": "^1.18.0",
     "express-slow-down": "^1.4.0",
     "express-winston": "^4.0.5",
+    "globals": "^15.12.0",
     "helmet": "^4.6.0",
     "inflection": "^1.13.1",
     "ioredis": "^4.27.6",
+    "jest": "^29.6.2",
```
Comment (apburnes) [inline package.json:39]: This should be in `devDependencies`

File: frontend/shared/ExpandableArea.test.jsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+
+import ExpandableArea from './ExpandableArea';
+
+describe('<ExpandableArea/>', () => {
+  it('renders', () => {
```
Comment (apburnes) [inline frontend/shared/ExpandableArea.test.jsx:8]: Can we add a test if it `isExpanded`?

**Review by apburnes — COMMENTED** [review]
This is a solid start for adding RTL tests. I put in just a few questions.

File: frontend/shared/ExpandableArea.test.jsx (hunk 2 above)
Comment (drewbo) [reply frontend/shared/ExpandableArea.test.jsx:8]: the expansion is still controlled by USWDS JS so, not easily.

Comment (apburnes) [reply frontend/shared/ExpandableArea.test.jsx:8]: Understood. Thanks!

**Review by apburnes — APPROVED** [review]

**Pull request merged**